### PR TITLE
Tests created for module

### PR DESCRIPTION
**Pull request by ep-93**
The code works, and I have it working locally with a plan utilising it in example. I need to merge this so I can refrence it and push a PR to mod-plat-env so it can be built in example via the pipeline rather than me applying anything locally.

The tests do not currently work.

Update.

Tests working, hardcoded subnet ID which needs to change.

### CODE REVIEW

File: test/unit-test/locals.tf [hunk 1]
```diff
@@ -1,38 +1,177 @@
...
+    }
+  }
+  ec2_test = {
+    tags = {
+      component = "test"
+    }
+
+    instance = {
+      disable_api_termination      = false
+      instance_type                = "t3.medium"
```
Comment (ewastempel) [inline test/unit-test/locals.tf:33]: we could use something cheaper instead

File: test/unit-test/security-groups.tf [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    cidr_blocks      = ["0.0.0.0/0"]
+    ipv6_cidr_blocks = ["::/0"]
+  }
+
+  tags = {
+    Name = "test"
+  }
+}
+
+data "aws_subnet" "private_subnets_a" {
```
Comment (davidkelliott) [inline test/unit-test/security-groups.tf:26]: Actually could you use this original data call as it is without hardcoding?

File: test/unit-test/locals.tf [hunk 3]
```diff
@@ -1,38 +1,177 @@
...
+          metadata_options_http_tokens = "optional"
+        }
+        ami_name  = "RHEL-6.10_HVM-*"
+        ami_owner = "309956199498"
+      }
+    }
+  }
+}
+
+data "aws_kms_key" "default_ebs" {
```
Comment (ewastempel) [inline test/unit-test/locals.tf:120]: perhaps  we could put all data into data.tf

File: test/unit-test/locals.tf [hunk 4]
```diff
@@ -1,38 +1,177 @@
...
+data "aws_iam_policy_document" "ec2_common_combined" {
+  source_policy_documents = [
+    data.aws_iam_policy_document.ec2_policy.json,
+  ]
+}
 
-  is_live       = [substr(terraform.workspace, length(local.application_name), length(terraform.workspace)) == "-production" || substr(terraform.workspace, length(local.application_name), length(terraform.workspace)) == "-preproduction" ? "live" : "non-live"]
-  provider_name = "core-vpc-${local.environment}"
+# create list of common managed policies that can be attached to ec2 instance profiles
+locals {
```
Comment (ewastempel) [inline test/unit-test/locals.tf:146]: multiple `local` annotations make it a bit messy

File: test/unit-test/secrets.tf [hunk 5]
```diff
@@ -14,3 +14,9 @@ data "aws_secretsmanager_secret_version" "environment_management" {
   provider  = aws.testing-ci-user
   secret_id = data.aws_secretsmanager_secret.environment_management.id
 }
+
+data "aws_vpc" "shared" {
```
Comment (ewastempel) [inline test/unit-test/secrets.tf:18]: this isn't a secret, so there is probably a better place for it

**Review by davidkelliott — DISMISSED** [review]

**Review by dms1981 — APPROVED** [review]

**Pull request merged**